src/cargo.rs

```diff
@@ -0,0 +1,140 @@
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+use std::io::prelude::*;
+use err::ProfError;
+use regex::Regex;
+use std::process::{Command, exit};
+use std::path::Path;
+
+/// Returns the closest ancestor path containing a `Cargo.toml`.
+///
+/// Returns `None` if no ancestor path contains a `Cargo.toml`, or if
+/// the limit of MAX_ANCESTORS ancestors has been reached.
+pub fn find_toml() -> Option<PathBuf> {
```

Comment:
The NoTargetDirectory error will require the project  to have a `target/` directory containing the binary to use cargo-profiler w/o --bin. However, the user still has the option of explicitly providing the path to the binary, if they have a non-standard output directory.